### PR TITLE
generating warning on adding existing alias

### PR DESCRIPTION
**Pull request by mittalswastik**
refering to - https://github.com/vector-im/riot-web/issues/3477
Now on adding an alias name would send an api request to check whether any room id for specified alias name exist or not, and if it does a warning is generated and the user is briefed about this before hand rather than informing the user ones he/she makes all the changes and then saves it. Also any change in alias name is checked, if existing or not and warnings are displayed accordingly.

### CODE REVIEW

File: src/components/views/room_settings/AliasSettings.js [hunk 1]
```diff
@@ -160,12 +172,42 @@ module.exports = React.createClass({
         }
     },
 
+    addAlias: function(alias) {
+         // add this alias to the domain to aliases dict
+        var domain = alias.replace(/^.*?:/, '');
```
Comment (t3chguy) [inline src/components/views/room_settings/AliasSettings.js]: use let/const where possible https://github.com/matrix-org/matrix-react-sdk/blob/master/code_style.md#ecmascript

Comment (mittalswastik) [inline src/components/views/room_settings/AliasSettings.js]: @t3chguy domain was already declared, I just shifted it from onAliasAdded to addAlias function. So I din't make any changes in those declaration.:) The only problem I am facing here is when both the warnings are generated that is onChange and onAdd which is highly unlikely then the orientation of the onAdd warning gets disturbed that is do not fall in the tabel cell instead is seen pretty off from its actual position.

Comment (t3chguy) [inline src/components/views/room_settings/AliasSettings.js]: I meant it generally, that was just the first one that jumped out, applies to https://github.com/matrix-org/matrix-react-sdk/pull/835/files#diff-26b9eb129e9736bd1599b3c40ff80f57R149 and more

**Comment by mittalswastik** [comment]
@t3chguy did you check this? Warning is a good solution but then adding an onChange functionality to it would create a difficult scenario,  suppose user gets a warning on adding new alias but skips this warning and edits previous existing alias hence there is a situation where user is allowed to skip this warning and move forward. This doesn't seem pretty for UI point of view. We could still customize it but that feels unnecessary instead is it really that important to have onChange on top of onAdd just to ease case of typo. Instead  ":matrix.org" should be preadded and user should not be asked for it again and again. And then typing just the name again ain't that big a task.

**Comment by lukebarnard1** [comment]
@mittalswastik, I take it you won't be making changes to this as per code review? It looks like it needs some delinting, make sure to run `npm run lintwithexclusions` to pick up on lint.

**Comment by jryans** [comment]
Thanks for making this contribution a while back. Since the code base has changed since this was opened, it no longer applies cleanly, and I don't think there's a need to keep it open in this state, as we can always find the code here again if needed. If you are still interested in pursuing this feature, please discuss with us in #riot-dev:matrix.org to find a good approach forward.

**Pull request closed without merging**